src/client/metadataApiDeploy.ts

```diff
@@ -352,6 +352,11 @@ export class MetadataApiDeploy extends MetadataTransfer<MetadataApiDeployStatus,
     const connection = await this.getConnection();
     // store for use in the scopedPostDeploy event
     this.orgId = connection.getAuthInfoFields().orgId;
+    if (this.components && !this.components.apiVersion && !this.components.sourceApiVersion) {
```

Comment:
@mshanemc I am not too sure about this assertion, but looking at component set class there is a function `getObject` that [establishes apiVersion](https://github.com/forcedotcom/source-deploy-retrieve/blob/sm/max-api-version-on-deploys/src/collections/componentSet.ts#L327) for the component. Is it possible that `getObject` has already been called before reaching this check?

Reply:
Within SDR, that's called during the creation of the package.xml, which is pretty late.

Technically, some consumer could do anything they want with it, but it's unlikely.

Reply:
what change are you requesting?

Reply:
Nothing to change if my suspicion was not confirmed. Thanks